### PR TITLE
Better fvtk.viz error when no VTK installed

### LINKED ISSUE #1238: Cryptic fvtk.viz error when no VTK installed
I still get this error when running the tracking_quickstart example:

    AttributeError: module 'dipy.viz.fvtk' has no attribute 'ren'

Would be good if there was some more explanatory error handling fencing this.

### PR DESCRIPTION
**Pull request by ghoshbishakh**
fixes #1238

### CODE REVIEW

File: dipy/utils/tripwire.py [hunk 1]
```diff
@@ -41,9 +41,14 @@ class TripWire(object):
...
     def __init__(self, msg):
         self._msg = msg
 
     def __getattr__(self, attr_name):
         ''' Raise informative error accessing attributes '''
         raise TripWireError(self._msg)
+
+    def __call__(self, *args, **kwargs):
+        ''' Raise informative error while calling '''
+        raise TripWireError(self._msg)
```
Comment (arokem) [inline dipy/utils/tripwire.py]: Could you please write a test that exercises this code?

**Comment by ghoshbishakh** [comment]
@arokem done!

File: dipy/viz/fvtk.py [hunk 2]
```diff
@@ -79,6 +79,9 @@
     msg = "Python VTK is not installed"
     warn(msg)
 
+    ren, have_ren, _ = optional_package('dipy.viz.window.ren',
```
Comment (arokem) [inline dipy/viz/fvtk.py]: Reading this again -- do you understand why the line above this one throws a warning, instead of raising an error?

Comment (ghoshbishakh) [inline dipy/viz/fvtk.py]: @arokem 
```
ren, have_ren, _ = optional_package('dipy.viz.window.ren',
                                        'Python VTK is not installed')
```
returns a TripWire object into `ren`

So calling `ren` will only raise an exception not the line 82.

Comment (arokem) [inline dipy/viz/fvtk.py]: I mean: if we don't have VTK installed, why don't we just raise an exception in the previous line (line 80)? Why do we bother going ahead and trying to do anything?

Comment (ghoshbishakh) [inline dipy/viz/fvtk.py]: @arokem I am not sure about that design decision. But there must be a reason for example it says `# Allow import, but disable doctests if we don't have vtk`, although I am not sure what that means. @Garyfallidis

Comment (arokem) [inline dipy/viz/fvtk.py]: @garyfallidis - I don't understand this. Is there any reason line 80 doesn't raise an error? If have_vtk is False, we're bound to hit an error sooner or later, no?

**Comment by arokem** [comment]
Merging. Will follow up with another PR if needed.

**Pull request merged**